### PR TITLE
feature(Authentication):Signup - Signin

### PR DESCRIPTION
**Pull request by katiamiyu**
### What does this PR do?
Create endpoints for signin and signup functionality
#### Description of Task to be completed?
Create authController
Create AuthRoute
Setup test environment for AuthController
Setup Postgresql as database
#### How should this be manually tested?
Clone the repository
run 'npm run build'
run 'npm start'
using postman navigate to this route for signup
post: http://localhost:3000/api/v1/auth/signup
{
"firstname":string",
"lastname": "string",
"email": "string",
"password":"string",
"isadmin":"boolean"
}
post: using postman navigate to this route for signin
http://localhost:3000/api/v1/auth/signin
And if you want to test the endpoints
run 'npm run build'
run 'npm run test'
{
"email": "string",
"password":"string"
}

### CODE REVIEW

**Comment by ajimae** [comment]
Your build tests are failing, you might want to check and fix all failing test.

File: lib/dataFile/controllers/authController.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
+"use strict";
```
Comment (ajimae) [inline lib/dataFile/controllers/authController.js]: Add this this file to your .gitignore file

File: lib/dataFile/dbCon/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"use strict";
```
Comment (ajimae) [inline lib/dataFile/dbCon/index.js]: git ignore this file too

File: lib/dataFile/decode.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+"use strict";
```
Comment (ajimae) [inline lib/dataFile/decode.js]: This file too, add to .gitignore

**Review by ajimae — CHANGES_REQUESTED** [review]
Add the entire lib/ folder to your `.gitignore` file.
Also, please fix failing tests

File: lib/dataFile/controllers/authController.js (hunk 1 above)
Comment (katiamiyu) [reply lib/dataFile/controllers/authController.js]: Done

File: lib/dataFile/dbCon/index.js (hunk 2 above)
Comment (katiamiyu) [reply lib/dataFile/dbCon/index.js]: done

File: lib/dataFile/decode.js (hunk 3 above)
Comment (katiamiyu) [reply lib/dataFile/decode.js]: Done

**Pull request merged**